src/libPMacc/include/eventSystem/transactions/Transaction.tpp

```diff
@@ -60,15 +47,43 @@ inline EventTask Transaction::getTransactionEvent( )
     return baseEvent;
 }
 
-void Transaction::operation( ITask::TaskType )
+void Transaction::operation( ITask::TaskType operation )
 {
-    if ( isAtomic == false )
-        baseEvent.waitForFinished( );
+    if ( operation == ITask::TASK_CUDA )
+    {
+        Manager &manager = Environment<>::get( ).Manager( );
+
+        ITask* baseTask = manager.getITaskIfNotFinished( this->baseEvent.getTaskId( ) );
+        if ( baseTask != NULL )
```

Comment:
This is not joint to be sure that `baseTask->getTaskType( )` is not called if `baseTask == NULL`. This can also done with short circuited operation but is more hard to see.